src/api/qualicharge/schemas/utils.py

```diff
@@ -0,0 +1,165 @@
+    created, entry_db = get_or_create(
+        session,
+        entry,
+        fields=fields,
+    )
+    if created or not update:
+        return created, entry_db
+
+    # Update
+    entry_db = entry_db.model_copy(
+        update=entry.model_dump(
+            exclude=fields if fields is None else fields | DB_TO_STATIC_EXCLUDED_FIELDS
+        )
+    )
+    session.add(entry_db)
+    session.commit()
+    return created, entry_db
+
+
+def save_statique(
+    session: Session, statique: Statique, update: bool = True
+) -> Statique:
+    """Save Statique instance to database."""
+    # Core schemas
+    _, pdc = save_schema_from_statique(
+        session, PointDeCharge, statique, fields={"id_pdc_itinerance"}, update=update
+    )
+    _, station = save_schema_from_statique(
+        session, Station, statique, fields={"id_station_itinerance"}, update=update
+    )
+    _, amenageur = save_schema_from_statique(
+        session, Amenageur, statique, update=update
+    )
+    _, operateur = save_schema_from_statique(
+        session, Operateur, statique, update=update
+    )
+    _, enseigne = save_schema_from_statique(session, Enseigne, statique, update=update)
+    _, localisation = save_schema_from_statique(
+        session, Localisation, statique, update=update
+    )
+
+    # Relationships
+    pdc.station_id = station.id  # type: ignore[attr-defined]
+    station.amenageur_id = amenageur.id  # type: ignore[attr-defined]
+    station.operateur_id = operateur.id  # type: ignore[attr-defined]
+    station.enseigne_id = enseigne.id  # type: ignore[attr-defined]
+    station.localisation_id = localisation.id  # type: ignore[attr-defined]
+
+    session.add(pdc)
+    session.add(station)
+    session.commit()
+
+    return Statique(
+        **pdc.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **station.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **operateur.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **amenageur.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **enseigne.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **localisation.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+    )
+
+
+def build_statique(session: Session, id_pdc_itinerance: str) -> Statique:
+    """Build Statique instance from database."""
+    pdc = session.exec(
+        select(PointDeCharge).where(
+            PointDeCharge.id_pdc_itinerance == id_pdc_itinerance
+        )
+    ).one()
+
+    return Statique(
+        **pdc.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **pdc.station.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **pdc.station.operateur.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **pdc.station.amenageur.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **pdc.station.enseigne.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+        **pdc.station.localisation.model_dump(exclude=DB_TO_STATIC_EXCLUDED_FIELDS),
+    )
```

Comment:
We should think of a more efficient way to handle fetching multiple PDC at once without exploding the number of DB requests.